utils/formatCountyDataVTwo.ts

```diff
@@ -0,0 +1,467 @@
+      chartType: ChartTypes.GENDER,
+      datasets: [],
+      displayLegend: false,
+      labels: [],
+      totalCount: 0,
+      customChartOptions: {
+        plugins: {}
+      }
+    },
+    raceEthGroup: {
+      chartType: ChartTypes.RACE_ETH,
+      datasets: [],
+      displayLegend: false,
+      labels: [],
+      customChartOptions: {
+        plugins: {}
+      }
+    },
+    raceEthNormGroup: {
+      chartType: ChartTypes.RACE_ETH_NORM,
+      datasets: [],
+      displayLegend: false,
+      labels: [],
+      customChartOptions: {
+        plugins: {}
+      }
+    }
+  }
+}
+
+const getPercentageData = (target: number, total: number) => {
+  return Math.round((target / total) * 100)
+}
+
+const getSumOfArray = (dataset: Array<number>): number => {
+  return dataset.reduce((acc: number, curr: number) => acc + curr)
+}
+
+const getTotalCount = (data: { [name: string]: number }): number => {
+  let total = 0
+  const keys = Object.keys(data)
+  for (const key of keys) {
+    const count = data[key]
+    if (count >= 0) total += count
+  }
+  return total
+}
+
+const getUpdatedCountyData = (
+  county: CountyData,
+  defaultFormattedData: FormattedCountyData
+) => {
+  const updatedData: any = {}
+  if (county) {
+    const {
+      case_totals: { age_group, gender, race_eth },
+      update_time,
+      source_url
+    } = county
+
+    if (update_time) {
+      updatedData.lastUpdatedAt = parseDateForYrMoDay(update_time)
+    }
+    if (source_url) {
+      updatedData.sourceUrl = source_url
+    }
+    if (age_group) {
+      updatedData.ageGroup = buildAgeChartData(
+        defaultFormattedData.ageGroup!,
+        age_group
+      )
+    }
+    if (gender) {
+      updatedData.genderGroup = buildGenderChartData(
+        defaultFormattedData.genderGroup!,
+        gender
+      )
+    }
+    if (race_eth) {
+      updatedData.raceEthGroup = buildRaceEthChartData(
+        defaultFormattedData.raceEthGroup!,
+        race_eth
+      )
+      updatedData.raceEthNormGroup = buildRaceEthNormalizedChartData(
+        defaultFormattedData.raceEthNormGroup!,
+        race_eth
+      )
+    }
+  }
+  return updatedData
+}
+
+const normalizeDataBy = (
+  populationNumber: number,
+  value: number,
+  toFixedNum?: number
+): number => {
+  const normNum = value / populationNumber
+  return toFixedNum ? parseFloat(normNum.toFixed(toFixedNum)) : normNum
+}
+
+const parseDateForYrMoDay = (date: string): string => {
+  return date.split('T')[0]
+}
+
+const sortLabelsAndValuesByValue = (data: any) => {
+  const sortByValue = (a: any, b: any) => {
+    if (a[1] < b[1]) return 1
+    if (a[1] > b[1]) return -1
+    return 0
+  }
+  return Object.entries(data).sort(sortByValue)
+}
+
+export default (
+  data: CountiesData,
+  allCounties: Array<string>
+): FormattedCountyData => {
+  const finalData: FormattedCountiesData = {}
+
+  for (const countyName of allCounties) {
+    const defaultFormattedData = clone(getDefaultFormattedData())
```

Comment:
I don't thinks clone is necessary here as getDefaultFormattedData already returns an initialized array.

Reply:
Yea clone is not necessary. Removed.